syntax = "proto3";

option go_package = "./spec";

service Control {
  rpc Command2Internal(RequestSync) returns (ResponseSync) {}
}

message RequestSync {
  Stations station = 1;
  enum State {
    UNKNOWN = 0;
    ON = 1;
    OFF = 2;
  }
  State state = 2;
}

message ResponseSync {
  enum Response {
    UNKNOWN = 0;
    SUCCESS = 1;
    FAILED = 2;
  }
  Response response = 1;
}

message Stations {
  enum StationId {
    unknown = 0;


    shinjuku_p1 = 1;
    shinjuku_p1 = 2;
    shinjuku_s1 = 3;
    shinjuku_s2 = 4;

    sakurajosui_p1 = 11;
    sakurajosui_p2 = 12;
    sakurajosui_s1 = 13;
    sakurajosui_s2 = 14;
    sakurajosui_s3 = 15;
    sakurajosui_s4 = 16;

    chofu_p1 = 21;
    chofu_p2 = 22;
    chofu_p3 = 23;
    chofu_p4 = 24;
    chofu_p5 = 25;
    chofu_s1 = 26;
    chofu_s2 = 27;
    chofu_s3 = 28;
    chofu_s4 = 29;

    hashimoto_s1 = 31;
    hashimoto_s2 = 32;

    hachioji_s1 = 41;
    hachioji_s2 = 42;

    wakabadai_p1 = 51;
    wakabadai_p2 = 52;
    wakabadai_s1 = 53;
    wakabadai_s2 = 54;
  }
  StationId stationId = 1;
}
